splitio/storage/pluggable.py

```diff
@@ -0,0 +1,195 @@
+
+        :param traffic_type_name: Traffic type to validate.
+        :type traffic_type_name: str
+
+        :return: True if the traffic type is valid. False otherwise.
+        :rtype: bool
+        """
+        return self._pluggable_adapter.get(self._traffic_type_prefix + traffic_type_name) != None
+
+    def kill_locally(self, split_name, default_treatment, change_number):
+        """
+        Local kill for split
+
+        :param split_name: name of the split to perform kill
+        :type split_name: str
+        :param default_treatment: name of the default treatment to return
+        :type default_treatment: str
+        :param change_number: change_number
+        :type change_number: int
+        """
+        split = self.get(split_name)
+        if not split:
+            return
+        if self.get_change_number() > change_number:
+            return
+        split.local_kill(default_treatment, change_number)
+        self._pluggable_adapter.set(self._prefix + split_name, split.to_json())
+        self.set_change_number(change_number)
+
+    def increase_traffic_type_count(self, traffic_type_name):
+        """
+        Increase by one the count for a specific traffic type name.
+
+        :param traffic_type_name: Traffic type to increase the count.
+        :type traffic_type_name: str
+        """
+        self._pluggable_adapter.increment(self._traffic_type_prefix + traffic_type_name, 1)
+
+    def decrease_traffic_type_count(self, traffic_type_name):
+        """
+        Decrease by one the count for a specific traffic type name.
+
+        :param traffic_type_name: Traffic type to decrease the count.
+        :type traffic_type_name: str
+        """
+        self._pluggable_adapter.decrement(self._traffic_type_prefix + traffic_type_name, 1)
+        if self._pluggable_adapter.get(self._traffic_type_prefix + traffic_type_name) == 0:
+            self._pluggable_adapter.delete(self._traffic_type_prefix + traffic_type_name)
+
+    def get_all_splits(self):
+        """
+        Return all the splits.
+
+        :return: List of all the splits.
+        :rtype: list
+        """
+        return self.get_all()
+
+    def is_valid_traffic_type(self, traffic_type_name):
+        """
+        Return whether the traffic type exists in at least one split in cache.
+
+        :param traffic_type_name: Traffic type to validate.
+        :type traffic_type_name: str
+
+        :return: True if the traffic type is valid. False otherwise.
+        :rtype: bool
+        """
+        return self.traffic_type_exists(traffic_type_name)
+
+    def put(self, split):
+        """
+        Store a split.
+
+        :param split: Split object.
+        :type split: splitio.models.split.Split
+        """
+        self._pluggable_adapter.set(self._prefix + split.name, split.to_json())
+        self._pluggable_adapter.increment(self._traffic_type_prefix + split.traffic_type_name, 1)
```

Comment:
You need to consider to decrement the traffic type of previous split in `_pluggable_adapter.get(self._prefix + split.name)`

Reply:
do you mean if the existing split has different traffic type before it gets updated?

Reply:
will need to check if split exist and decrement the old tt and increment new tt